### PR TITLE
Multi-currency addition with order.paid webhook

### PR DESCRIPTION
**Pull request by ramth05**
Added code changes for JIRA (PAY-1293: Multi-currency support, PAY-1330: Webhook consumption ).

### CODE REVIEW

File: razorpay/controllers/front/validation.php [hunk 1]
```diff
@@ -60,7 +60,33 @@ public function postProcess()
...
+            {
+
+                Logger::addLog("Payment Failed for Order# ".$cart->id.". Razorpay payment id: ".$paymentId. "Error: ". $error, 4);
+
+                echo 'Error! Please contact the seller directly for assistance.</br>';
+                echo 'Order Id: '.$cart->id.'</br>';
+                echo 'Razorpay Payment Id: '.$paymentId.'</br>';
+                echo 'Error: '.$e->getMessage().'</br>';
+
+                exit;
```
Comment (vikas22) [inline razorpay/controllers/front/validation.php]: Why `exit`? Can't we use `return`?

File: razorpay/razorpay-webhook.php [hunk 2]
```diff
@@ -20,6 +20,7 @@ class RZP_Webhook
      */
     protected $api;
 
+
```
Comment (ShivamYuvraj) [inline razorpay/razorpay-webhook.php]: extra space

File: razorpay/razorpay-webhook.php [hunk 3]
```diff
@@ -49,6 +50,7 @@ public function process()
     {
         $post = file_get_contents('php://input');
 
+
```
Comment (ShivamYuvraj) [inline razorpay/razorpay-webhook.php]: extra space

File: razorpay/razorpay-webhook.php [hunk 4]
```diff
@@ -133,38 +135,111 @@ protected function paymentAuthorized(array $data)
      */
     protected function orderPaid(array $data)
     {
-        //reference_no (ps order id) should be passed in payload
-        $orderId = $data['payload']['payment']['entity']['notes']['reference_no'];
+        // reference_no (prestashop_cart_id) should be passed in payload
+        $cartId = $data['payload']['payment']['entity']['notes']['prestashop_cart_id'];
+
+        // check if a order already present for this cart
+        $cart = new Cart($cartId);
```
Comment (ShivamYuvraj) [inline razorpay/razorpay-webhook.php]: this looks redundant, we are creating a card using cart id and then get getting id from card. Are two ids different

**Comment by ramth05** [comment]
> Please make sure that we send 400 response code whenever there is a validation error or signature failed in incoming webhook request

ok.

File: razorpay/razorpay-webhook.php (hunk 2 above)
Comment (ramth05) [reply razorpay/razorpay-webhook.php]: removed

File: razorpay/razorpay-webhook.php (hunk 3 above)
Comment (ramth05) [reply razorpay/razorpay-webhook.php]: fixed

**Pull request merged**